concent_api/api_testing_common.py

```diff
@@ -163,33 +164,32 @@ def _print_data(data, url):
     validate_content_type(response.headers['Content-Type'], expected_content_type)
     validate_response_message(response.content, expected_message_type, private_key, public_key)
     print()
-    if response.status_code in [202, 204]:
+    content_type = response.headers['Content-Type']
+    if 'text/html' in content_type:
         return None
-    else:
+    elif 'json' in content_type:
+        return json.loads(response._content)
+    elif 'application/octet-stream' in content_type:
         return try_to_decode_golem_message(private_key, public_key, response.content)
+    else:
+        raise UnexpectedResponse(f'Unexpected response content_type. Responses content type is {content_type}.')
 
 
 def _print_response(private_key, public_key, response):
     if response.content is None:
         print('RESPONSE: <empty>')
-    elif len(response.content) != 0:
-        _print_message_from_response(private_key, public_key, response)
     else:
-        print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
-        if response.text not in ['', None]:
-            print('RAW RESPONSE (text): {}'.format(response.text))
-
-
-def _print_message_from_response(private_key, public_key, response):
-    print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
-    print('MESSAGE:')
-    print('Concent-Golem-Messages-Version = {}'.format(response.headers['concent-golem-messages-version']))
-    if response.headers['Content-Type'] == 'application/octet-stream':
-        _print_message_from_stream(private_key, public_key, response.content)
-    elif response.headers['Content-Type'] == 'application/json':
-        _print_message_from_json(response)
-    else:
-        print('ERROR: Unexpected content-type of response message')
+        print(f'STATUS: {response.status_code} {http.client.responses[response.status_code]}')
+        print('MESSAGE:')
+        print(f'Concent-Golem-Messages-Version = {response.headers["concent-golem-messages-version"]}')
+        if 'application/octet-stream' in response.headers['Content-Type']:
+            _print_message_from_stream(private_key, public_key, response.content)
+        elif 'application/json' in response.headers['Content-Type']:
+            _print_message_from_json(response)
+        elif 'text/html' in response.headers['Content-Type']:
+            pass
```

Comment:
I don't think it is necessary.. in all cases I checked message was empty- message.text was "". I looked for something useful to print and I didn't found. 

Second thing- this function is testing tool, used only by developers in tests. It always prints status code and headers. And if something goes wrong we have all server logs anyway..

Reply:
If header with `Concent-Type` value is printed then its ok.

Reply:
I think that it might be useful to print the beginning and the end of the response. The thing is that in case of Concent we almost never expect HTML so if you get it, the first thing you want is to see what you got. Not the whole thing but a part of it might be useful.

That said, it's not a big deal. As you say - it's a development tool so you can always just repeat the test and see in your debugger what you got. So it can stay as is if you prefer.

But one thing: there should be a comment explaining this. You don't have to say what you do (that's obvious) but always say why.